specification/sections/02-providers.md

```diff
@@ -26,6 +26,22 @@ Providers are the "translator" between the flag evaluation calls made in applica
 provider.getMetadata().getName(); // "my-custom-provider"
 ```
 
+#### Requirement 2.1.2
+
+> The provider `metadata` member or accessor **MAY** define an `allFlagMetadata` field or accessor of type collection of [flag_metadata](../types.md#flag-metadata), which identifies all available flag keys in the provider.
+
+```typescript
+provider.getMetadata().getAllFlagMetadata(); // [{"key": "featureA", "type": "boolean"}, {"key": "featureB", "type": "string"}]
+```
+
+#### Condition 2.1.3
+
+> The implementing language has an asynchronous support.
+
+##### Conditional Requirement 2.1.3.1
+
+> The accessor for `allFlagMetadata` **SHOULD** be asynchronous.
```

Comment:
This requirement would change how we treat metadata, at least in the JS SDK. Currently, metadata is static data that conforms to an interface.

https://github.com/open-feature/js-sdk/blob/main/packages/shared/src/provider/provider.ts#L33-L35

Reply:
That is a good point; having it on the provider interface directly removes the condition. I wouldn't want to change the static nature of metadata with this change, and I think there's a reasonable expectation that this call would be dynamic for each call.

I'm going to add it instead to the provider interface, as well as expose it to the client and see how that looks.